ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,220 @@
+# 47. GitOps Onboarding Redesign
+
+Date: 2025-07-09
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, Konflux offers two distinct onboarding paths that create friction and complexity for developers:
+
+### UI Onboarding
+- Very easy to use with good validation and feedback
+- Handles secrets, build config, environments through web interface
+- Provides immediate feedback but doesn't align with GitOps principles
+
+### GitOps Onboarding
+- Requires manually authoring YAML across multiple files
+- Validation happens during GitLab CI runs, creating tension between Konflux and GitLab CI workflows
+- Secrets must still be created via UI, breaking the GitOps flow
+- Onboarding is slow and often requires multiple failed PRs to converge due to trial-and-error approach
+
+### Current Pain Points
+1. **Fragmented experience**: Developers must context-switch between UI and GitOps approaches
+2. **Validation delays**: Schema validation only occurs during CI runs, not locally
```

Comment:
This isn't true of the existing RH gitops repo. Run `tox` (or write a VSCode plugin to run `tox`).

Reply:
I have done that, and it resulted in merge conflict hell - I think its been improved since, but having a monorepo makes merge conflicts much more likely than distributed individual repos

Reply:
Update the pain points to reflect that. That's not what the bullet says.